engine/src/main/resources/org/camunda/bpm/engine/impl/mapping/entity/Job.xml

```diff
@@ -201,6 +201,13 @@
         )
       </if>
 
+      <if test="parameter.jobPriorityMin != null">
+        and RES.PRIORITY_ &gt;= #{parameter.jobPriorityMin}
+      </if>
+      <if test="parameter.jobPriorityMax != null">
+        and RES.PRIORITY_ &lt;= #{parameter.jobPriorityMax}
+      </if>
```

Comment:
The advantage of using `0` and `MAX_LONG` is that it is much easier to validate the configuration. We also have default values for `batchJobPriority` and `historyCleanupPriority` of `0`. We would need null checks whenever we compare priorities if we use `null` as default. Writing this, I realized we should probably add null checks anyways as `null` is still a possible value or use `long` instead of `Long` for the properties.

To improve the SQL query I would suggest we only pass the `jobPriorityMin` and `jobPriorityMax` parameters if they are not the default values. Alternatively, we only pass the parameters if `jobExecutorAcquireByPriority` is set to `true`. This way the parameters would be null if the feature is not used.

What do you think?

Reply:
> A range has an upper and a lower bound, both are optional

I can read that both bounds should be optional in the feature requirements, meaning that it should also be possible to set a lower *xor* upper bound. This means `jobPriorityMin=77`, and `jobPriorityMax=null` should be possible, which is even more efficient than `jobPriorityMax=MAX_LONG`. However, right now, this doesn't work either since the null checks are missing, leading to `null` pointer exceptions.

Regarding the default behavior of the query: I think solution one contradicts what I've written above. I prefer solution two: skip both query conditions when `jobExecutorAcquireByPriority = false`. However, I would still change the default values to `jobPriorityMin=null` and `jobPriorityMax=null` since this is effectively jobPriorityMin=0` and `jobPriorityMax=MAX_LONG` without introducing extra query conditions. When a user only wants to use one of the bounds, the other bound condition is not applied to the query.

Reply:
I agree that the query should only become more complex if the feature is used. `jobExecutorAcquireByPriority` should be independent: I can use the priority range without wanting the jobs to acquired in the order of their priority. I wouldn't mix these two configuration options.

Reply:
Makes sense. So I will:
* make `null` the default for both properties
* add null checks where the properties are used (it's actually not a lot of places where we need them)
* leave the query and parameter map as it is (with null being the default, the null checks in the query will actually work and the resulting query should be less complex)